nuxeo-core/nuxeo-core-bulk/src/main/java/org/nuxeo/ecm/core/bulk/action/SetSystemPropertiesAction.java

```diff
@@ -52,14 +63,37 @@ public Topology getTopology(Map<String, String> options) {
 
     public static class SetSystemPropertyComputation extends AbstractBulkComputation {
 
+        public static final String NOTIFY = "param-notify";
+
         public SetSystemPropertyComputation() {
             super(ACTION_NAME);
         }
 
         @Override
         protected void compute(CoreSession session, List<String> ids, Map<String, Serializable> properties) {
-            ids.forEach(id -> properties.forEach((k, v) -> session.setDocumentSystemProp(new IdRef(id), k, v)));
+            Map<String, Serializable> actualProperties = new HashMap<>(properties);
+            String eventId = (String) actualProperties.remove(NOTIFY);
+            Collection<DocumentRef> refs = ids.stream().map(IdRef::new).collect(Collectors.toList());
+            ids.forEach(id -> actualProperties.forEach((k, v) -> session.setDocumentSystemProp(new IdRef(id), k, v)));
```

Comment:
Why not applying `forEach` on `refs` directly ?
Furthermore, @efge has made changes in `SetPropertyComputation` to better handle error, does it apply there ?

Reply:
yes we should apply the same logic as in setProperties here, write access error will be very common since we allow non admin to run bulk operation